### PR TITLE
fix: Wraps select statement inside insert with parenthesis

### PR DESCRIPTION
**Pull request by Giovannini**
Fix issue: https://github.com/brianc/node-sql/issues/360

### CODE REVIEW

File: lib/dialect/postgres.ts [hunk 1]
```diff
@@ -392,9 +392,11 @@ export class Postgres extends Dialect {
         const paramNodes = insertNode.getParameters();
         if (paramNodes.length > 0) {
             const paramText = paramNodes
-                .map((paramSet) => {
-                    return paramSet.map((param) => this.visit(param)).join(', ');
-                })
+                .map((paramSet) => paramSet.map((param) => {
+                    const value = this.visit(param).toString()
+                    if (value.startsWith('SELECT')) return `(${value})`
```
Comment (Giovannini) [inline lib/dialect/postgres.ts:397]: This solution is very naïve. There may be a more elegant one to deal with all kind of expression but I do not have sufficient knowledge over the library. Please tell me if you see a better option.

**Comment by charsleysa** [comment]
Hi @Giovannini

Thanks for the PR.

One question I have is the use case for such a feature as I believe this can already be achieved by using the `.subQuery()` capability.

If using `.subQuery()` is not able to achieve the desired functionality, I am happy to consider this PR.

Please do let me know if this PR is still necessary after considering `.subQuery()`

**Comment by Giovannini** [comment]
Hello @charsleysa

Yes, `.subQuery()` was actually all I needed. Thank you!

Is there a place I may add some documentation about it as I did not know it even existed ?

**Comment by charsleysa** [comment]
There currently isn't any many docs, most functionality can be seen in the tests.

If you're open to contributing docs, create a docs folder and add markdown docs. I'd be more than happy to accept documentation PRs.

As you have noted that `.subQuery()` has the functionality you desire, I am going to close this PR.

**Pull request closed without merging**